lib/identifiers/orcid.rb

```diff
@@ -14,7 +18,7 @@ def self.valid?(str)
     end
 
     def self.calculate_digit(str)
-      return unless str =~ REGEX
+      return unless str.to_s =~ REGEX
```

Comment:
Do you need the `to_s` here as `=~` should work on quite a wide variety of objects?

```ruby
[1] pry(main)> nil =~ /foo/
=> nil
[2] pry(main)> [] =~ /foo/
=> nil
[3] pry(main)> {} =~ /foo/
=> nil
```